pkg/component/autoscaling/gardenercustommetrics/component.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gardenercustommetrics implements the gardener-custom-metrics seed component (aka GCMx).
+// For details, see the GardenerCustomMetrics type.
+package gardenercustommetrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/component/autoscaling/gardenercustommetrics/kubeobjects"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	secretsutils "github.com/gardener/gardener/pkg/utils/secrets"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+// ComponentName is the component name.
+const ComponentName = componentBaseName
+
+// GardenerCustomMetrics manages an instance of the gardener-custom-metrics component (aka GCMx). The component is
+// deployed on a seed, scrapes the metrics from all shoot kube-apiserver pods, and provides custom metrics by
+// registering as APIService at the custom metrics extension point of the seed kube-apiserver.
+// For information about individual fields, see the NewGardenerCustomMetrics function.
+type GardenerCustomMetrics struct {
+	namespaceName      string
+	containerImageName string
+	kubernetesVersion  *semver.Version
+
+	seedClient     client.Client
```

Comment:
Rename this to `client` only: https://github.com/gardener/gardener/commit/0754f0e742746f3251428e55820d8821cfbac717#diff-86c90c029b20e859adf4e91611870a11f58962ee2602a2f439bf99b0a1fe9338R45

Reply:
GCMx connects to the API servers of multiple clusters, which creates a possible ambiguity if the the common "client" name is used.

However, I understand that the use of a "client" name is a mandatory code consistency requirement. Therefore, renaming as requested.

Reply:
> GCMx connects to the API servers of multiple clusters, which creates a possible ambiguity if the the common "client" name is used.

I wouldn't say that this is the case. Maybe you are confused by the fact that the clients in the gardener-custom-metrics repo itself are connecting to different type of clusters (Seed cluster, Shoot cluster)? However this GCMx component only interacts with the Seed cluster for now - creates/deletes MR.

If we have different clients in our component (Seed client and Shoot client), then as suggested in https://github.com/gardener/gardener/pull/9570#discussion_r1562274350 we have to rather use the terminology runtime/target client. However, this is not the case for the GCMx component, hence we can use only `client.`

Reply:
Ah, you're right! Objection retracted.